src/test/java/jewellery/inventory/unit/service/ProductServiceTest.java

```diff
@@ -0,0 +1,264 @@
+    user.setResourcesOwned(List.of(resourceInUser));
+    productRequestDto.setProductsContent(List.of(testProduct.getId()));
+
+    assertThrows(
+        ProductNotFoundException.class, () -> productService.createProduct(productRequestDto));
+  }
+
+  @Test
+  void testCreateProductShouldSetContentProduct() {
+
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    when(resourceInUserRepository.findByResourceIdAndOwnerId(pearl.getId(), user.getId()))
+        .thenReturn(Optional.of(resourceInUser));
+    user.setResourcesOwned(List.of(resourceInUser));
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+
+    productRequestDto.setProductsContent(List.of(testProduct.getId()));
+
+    ProductResponseDto response = new ProductResponseDto();
+    when(productMapper.mapToProductResponseDto(any())).thenReturn(response);
+
+    ProductResponseDto actual = productService.createProduct(productRequestDto);
+
+    assertEquals(response, actual);
+  }
+
+  @Test
+  void testCreateProductShouldThrowWhenProductNotFound() {
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    user.setResourcesOwned(List.of(resourceInUser));
+    productRequestDto.setProductsContent(List.of(UUID.randomUUID()));
+
+    assertThrows(
+        ProductNotFoundException.class, () -> productService.createProduct(productRequestDto));
+  }
+
+  @Test
+  void testCreateProductShouldThrowExceptionWhenResourceNotFound() {
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+
+    assertThrows(
+        ResourceInUserNotFoundException.class,
+        () -> productService.createProduct(productRequestDto));
+  }
+
+  @Test
+  void testCreateProductGetUserShouldThrowExceptionIfUserNotExist() {
+    assertThrows(
+        UserNotFoundException.class, () -> productService.createProduct(productRequestDto));
+  }
+
+  @Test
+  void testGetProductShouldThrowWhenProductNotFound() {
+    UUID fakeId = UUID.fromString("58bda8d1-3b3d-4319-922b-f5bb66623d71");
+    assertThrows(ProductNotFoundException.class, () -> productService.getProduct(fakeId));
+  }
+
+  @Test
+  void testGetProductWhenProductFound() {
+
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+
+    ProductResponseDto response = new ProductResponseDto();
+    when(productMapper.mapToProductResponseDto(any())).thenReturn(response);
+
+    ProductResponseDto actual = productService.getProduct(testProduct.getId());
+
+    assertEquals(response, actual);
+  }
+
+  @Test
+  void testGetAllProducts() {
+
+    List<Product> products = Arrays.asList(testProduct, new Product(), new Product());
+
+    when(productRepository.findAll()).thenReturn(products);
+
+    List<ProductResponseDto> responses = productService.getAllProducts();
+
+    assertEquals(products.size(), responses.size());
+  }
```

Comment:
These may be obsolete if we are testing them properly in the intergation/feature tests.

Reply:
I did more assertions in integration test